client/packages/common/src/ui/components/inputs/Filters/DateFilter.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { FC } from 'react';
+import { RangeObject, useUrlQuery, RANGE_SPLIT_CHAR } from '@common/hooks';
+import { DatePickerInput, DateTimePickerInput } from '@common/components';
+import { FILTER_WIDTH, FilterDefinitionCommon } from './FilterMenu';
+import { DateUtils, useFormatDateTime } from '@common/intl';
+
+export interface DateFilterDefinition extends FilterDefinitionCommon {
+  type: 'date' | 'dateTime';
+  range?: RangeOption;
+}
+
+type RangeOption = 'from' | 'to';
+
+// Matches range strings for either just dates or date/times, with "_" as the
+// splitting character. Both the start date and end date are optional, but
+// must be the same (date / dateTime) if both present
+//
+// E.g the following will all match
+// 2023-10-02 03:10_2023-10-03 02:10
+// 2023-10-02_2023-10-03
+// 2023-10-02 03:10_
+// _2023-10-03
+const dateRangeRegex = new RegExp(
+  `^(\\d{4}-\\d{2}-\\d{2})?_(\\d{4}-\\d{2}-\\d{2})?|(\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2})?${RANGE_SPLIT_CHAR}(\\d{4}-\\d{2}-\\d{2} \\d{2}:\\d{2})?$`
+);
+
+export const DateFilter: FC<{ filterDefinition: DateFilterDefinition }> = ({
+  filterDefinition,
+}) => {
+  const { type, urlParameter, name, range } = filterDefinition;
```

Comment:
The filter looks for the presence of the `range` prop to know whether it is a stand-alone value or if it's part of a range group. In the latter case, the value of range (`from` / `to`) determines the subsequent behaviour (reading and writing the url parameter)